### ISSUE TITLE
`ProcessBuilder.lines` conflicts with `StringOps.lines`

### ISSUE THREAD

**Issue by kriomant** [sbt/sbt#819]
Write `"string".lines` in `build.scala` and you'll get compilation error:

```
[error] build.scala:68: type mismatch;
[error]  found   : java.lang.String("string")
[error]  required: ?{val lines: ?}
[error] Note that implicit conversions are not applicable because they are ambiguous:
[error]  both method augmentString in object Predef of type (x: String)scala.collection.immutable.StringOps
[error]  and method stringToProcess in trait ProcessExtra of type (command: String)sbt.ProcessBuilder
[error]  are possible conversion functions from java.lang.String("string") to ?{val lines: ?}
```

I can resolve conflict manually, but is very unexpected and confusing that standard `lines` method changed its meaning from "split text into lines" to "start program". I would expected something like `"string".!.lines` for this.

**Comment by som-snytt** [comment]
I just happened to notice a @paulp comment on the scala issue to deprecate linesIterator in favor of lines, that because of (standard lib) ProcessBuilder.lines, you almost want a different name on StringOps. (Or vice-versa, of course.)  Personally, I'd say PB.toLines and SO.asLines.

**Comment by kriomant** [comment]
`PB.toLines` is also bad, because string can be implicitly converted to `PB`. Do you really expect (just looking at code) that "cat\ndog".toLines does not split string into lines, but **starts** command? Starting command **must** be explicit.

**Comment by jsuereth** [comment]
I'd prefer we just use the syntax Proces("cat\ndog").lines.  Maybe we can deprecate the string -> ProcessBuilder implicit....

**Comment by kriomant** [comment]
Just idea: use custom string interpolation, like `run"program $arg1 $arg2".lines`

**Comment by paulp** [comment]
> Just idea: use custom string interpolation, like run"program $arg1 $arg2".lines

That's a great idea. Regardless, I'd say the String -> ProcessBuilder implicit should be deprecated.

**Comment by eed3si9n** [comment]
I'm assigning this to myself. If we were to deprecate String -> ProcessBuilder, do we have to go through the `@deprecated` cycle in 0.14?

**Comment by harrah** [comment]
Thanks for taking this.  Yes, we can't have `"asdf".!` break without a deprecation cycle.  However, if `"asdf".lines` doesn't even work right now, that can probably just be removed, for example.

**Comment by SethTisue** [comment]
hooray for deprecating the String -> ProcessBuilder implicit

**Comment by harrah** [comment]
Fixed by #913.
